### ISSUE TITLE
[Request]: Show download progress in the taskbar

### ISSUE THREAD

**Issue by DeepChirp** [CollapseLauncher/Collapse#502]
### Is your request related to a problem?

No

### Describe your proposed solution

I think the download progress can be displayed in the taskbar, so that the download progress can be quickly judged through the taskbar.

### Describe alternatives you've considered

Maybe not.

### Additional context

We may be able to use `TaskbarItemInfo.ProgressState` and `TaskbarItemInfo.ProgressValue` to achieve this functionality. See:
https://learn.microsoft.com/en-us/dotnet/api/system.windows.shell.taskbariteminfo.progressstate and https://learn.microsoft.com/en-us/dotnet/api/system.windows.shell.taskbariteminfo.progressvalue. Regardless of whether you adopt this suggestion, I am grateful.

**Comment by shatyuka** [comment]
We can implement this feature, but it may not have a high priority.
Also we use WinUI 3 as the UI framework, not WPF.
https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-itaskbarlist3-setprogressvalue

**Comment by DeepChirp** [comment]
>  Also we use WinUI 3 as the UI framework, not WPF.

Sorry, I don't know about these frameworks. I just saw it using C#.

**Comment by bagusnl** [comment]
Hi, after talking a bit about this with @neon-nyan, it's possible to implement this function as addition into our TrayIcon class. But this requires a lot of events code from InstallManager to handle cases like multiple download install, task bar status rate limit, etc.

TL;DR: It's possible but might take awhile due to the complexity of our code

**Comment by bagusnl** [comment]
Bump, any dev(s) want to dibs on this implementation?

**Comment by shatyuka** [comment]
working on it

**Comment by shatyuka** [comment]
Try this CI build: https://nightly.link/CollapseLauncher/Collapse/actions/runs/12733592000
